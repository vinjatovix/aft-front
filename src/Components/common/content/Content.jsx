import React from "react";
import PropTypes from "prop-types";
import { ActionButton } from "../../ui/buttons/ActionButton";
import { modalsStateType } from "../../../PropTypes";

export const Content = ({ children, modals, actions, isEditor, isOpen }) => {
  return (
    <div className={modals.blur ? "content blur" : "content"}>
      <div className="content_header">
        {(isEditor || isOpen) && (
          <ActionButton text="AÃ±adir" type="bg-green" action={actions.add} />
        )}
      </div>
      {children}
    </div>
  );
};

Content.propTypes = {
  children: PropTypes.node.isRequired,
  modals: modalsStateType,
  actions: PropTypes.shape({
    add: PropTypes.func.isRequired,
  }).isRequired,
  isEditor: PropTypes.bool,
};

Content.defaultProps = {
  isEditor: false,
};
