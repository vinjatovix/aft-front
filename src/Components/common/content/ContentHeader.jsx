import React from "react";
import PropTypes from "prop-types";

export const ContentHeader = ({ book, title, count }) => {
  return (
    <>
      {book && (
        <>
          <img
            className="portrait"
            src={
              book.img
                ? book.img
                : "https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg"
            }
            alt="book"
          />
          <h1>
            {book.author}-{book.name}
          </h1>
          <h2>{book.description}</h2>
        </>
      )}

      <h3 data-testid={title}>
        {title} {count}
      </h3>
    </>
  );
};

ContentHeader.propTypes = {
  book: PropTypes.shape({
    name: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    img: PropTypes.string,
    description: PropTypes.string.isRequired,
  }),
  title: PropTypes.string.isRequired,
  count: PropTypes.number,
};

ContentHeader.defaultProps = {
  book: null,
  count: null,
};
