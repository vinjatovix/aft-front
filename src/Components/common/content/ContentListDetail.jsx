import React from "react";
import PropTypes from "prop-types";
import { AdminButtons } from "../../ui/admin/AdminButtons";

export const ContentListDetail = ({ actions, item, token, isAdmin, type }) => {
  const scene = (
    <p>
      {item.name}-{item.location} {item.time}: {item.description}.
    </p>
  );
  const character = (
    <p>
      {item.name}: {item.description}. {item.center}
    </p>
  );

  return (
    <>
      <div
        onClick={() => {
          actions.setDataDetail(item);
        }}
      >
        {type === "scene" && scene}
        {type === "character" && character}
      </div>
      {isAdmin && (
        <AdminButtons type="row" actions={actions} item={item} token={token} />
      )}
    </>
  );
};

ContentListDetail.propTypes = {
  type: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setDataDetail: PropTypes.func.isRequired,
    add: PropTypes.func.isRequired,
    close: PropTypes.func.isRequired,
    delete: PropTypes.func.isRequired,
    edit: PropTypes.func.isRequired,
    detail: PropTypes.func.isRequired,
    refresh: PropTypes.func.isRequired,
  }).isRequired,
  item: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    _id: PropTypes.string.isRequired,
  }).isRequired,
  token: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool,
};

ContentListDetail.defaultProps = {
  isAdmin: false,
};
