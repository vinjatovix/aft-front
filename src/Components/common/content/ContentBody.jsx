import React from "react";
import PropTypes from "prop-types";
import { ListItem } from "../ListItem";
import { ContentListDetail } from "./ContentListDetail";

export const ContentBody = ({ content, isAdmin, actions, token, type }) => {
  return (
    <ul className="content-list">
      {content &&
        content.map((item) => (
          <ListItem key={item._id} className="content-list_item" {...item}>
            <ContentListDetail
              type={type}
              isAdmin={isAdmin}
              token={token}
              actions={actions}
              item={item}
            />
          </ListItem>
        ))}
    </ul>
  );
};

ContentBody.propTypes = {
  content: PropTypes.array,
  isAdmin: PropTypes.bool,
  actions: PropTypes.object.isRequired,
  token: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

ContentBody.defaultProps = {
  content: [],
  isAdmin: false,
};
