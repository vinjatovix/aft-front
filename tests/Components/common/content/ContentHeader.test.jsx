import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { ContentHeader } from "../../../../src/Components/common/content/ContentHeader";

describe("ContentHeader Component", () => {
  it("should match snapshot", async () => {
    const { container } = render(
      <Router>
        <ContentHeader
          title="Escenas:"
          count={3}
          book={{
            name: "Obra",
            author: "Autor",
            description: "DescripciÃ³n",
          }}
        />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
  it("should show title Escenas: 3", async () => {
    render(
      <Router>
        <ContentHeader title="Escenas:" count={3} />
      </Router>
    );
    const h1 = screen.getByTestId("Escenas:");

    expect(h1.innerHTML).toContain("Escenas: 3");
  });
});
