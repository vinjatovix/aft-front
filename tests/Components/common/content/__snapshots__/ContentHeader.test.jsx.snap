// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHeader Component should match snapshot 1`] = `
<div>
  <img
    alt="book"
    class="portrait"
    src="https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg"
  />
  <h1>
    Autor
    -
    Obra
  </h1>
  <h2>
    Descripci√≥n
  </h2>
  <h3
    data-testid="Escenas:"
  >
    Escenas:
     
    3
  </h3>
</div>
`;
